src/lib/storage/proposal/partition_creator.rb

```diff
@@ -281,14 +281,27 @@ def create_partition(vol, partition_id, free_slot)
           log.info("Creating partition for #{vol.mount_point} with #{vol.size}")
           disk = ::Storage::Disk.find(devicegraph, free_slot.disk_name)
           ptable = disk.partition_table
-          if logical_partition_preferred?(ptable)
-            create_extended_partition(disk, free_slot.slot.region) unless ptable.has_extended
+
+          logical = false
+          if space_inside_extended?(free_slot)
+            logical = true
+          elsif logical_partition_preferred?(ptable)
+            # It's not possible to have more than one extended partition,
+            # even if logical partitions are preferred
+            if !ptable.has_extended
+              create_extended_partition(disk, free_slot.slot.region)
+              logical = true
+            end
+          end
```

Comment:
I'm wondering if this can happen:

[ small_gap | p1 | p2 | p3 | big_gap ]

Then create a new partition -> fits in small_gap -> needs extended -> small_gap becomes extended, big_gap is lost.

Hm...